update.c

```diff
@@ -187,6 +191,89 @@ do_run_dialog (GtkListStore *list_store,
 }
 
 static void
+update_background_image_path (XdgDirEntry *old_entries,
+                              XdgDirEntry *new_entries)
+{
+  GSettings *settings = NULL;
+
+  XdgDirEntry *old_entry = NULL;
+  XdgDirEntry *new_entry = NULL;
+
+  char *old_uri = NULL;
+  char *new_uri, *new_path, *old_path;
+  char **split_path = NULL;
+
+  settings = g_settings_new (WP_PATH_ID);
+
+  old_uri = g_settings_get_string (settings, WP_URI_KEY);
+  if (old_uri == NULL)
+    {
+      g_warning ("No old image URI set. Ignoring background image change.");
+      goto out;
+    }
+
+  old_path = g_filename_from_uri (old_uri, NULL, NULL);
```

Comment:
`g_filename_from_uri()` can fail when the URI can't be mapped to a path; for instance when the background is the default value on Endless (`eos:///default`).
So you should just bail when `old_path == NULL`